src/test/java/org/elasticsearch/test/disruption/LongGCDisruption.java

```diff
@@ -23,76 +24,51 @@
 import java.util.HashSet;
 import java.util.Random;
 import java.util.Set;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.regex.Pattern;
 
+/**
+ * Suspends all threads on the specified node in order to simulate a long gc.
+ */
 public class LongGCDisruption extends SingleNodeDisruption {
 
-    volatile boolean disrupting;
-    volatile Thread worker;
-
-    final long intervalBetweenDelaysMin;
-    final long intervalBetweenDelaysMax;
-    final long delayDurationMin;
-    final long delayDurationMax;
-
-
-    public LongGCDisruption(Random random) {
-        this(null, random);
-    }
-
-    public LongGCDisruption(String disruptedNode, Random random) {
-        this(disruptedNode, random, 100, 200, 300, 20000);
-    }
+    private final static Pattern[] unsafeClasses = new Pattern[]{
+            // logging has shared JVM locks - we may suspend a thread and block other nodes from doing their thing
+            Pattern.compile("Logger")
+    };
 
-    public LongGCDisruption(String disruptedNode, Random random, long intervalBetweenDelaysMin,
-                            long intervalBetweenDelaysMax, long delayDurationMin, long delayDurationMax) {
-        this(random, intervalBetweenDelaysMin, intervalBetweenDelaysMax, delayDurationMin, delayDurationMax);
-        this.disruptedNode = disruptedNode;
-    }
+    protected final String disruptedNode;
+    private final AtomicReference<Set<Thread>> suspendedThreadsRef;
 
-    public LongGCDisruption(Random random,
-                            long intervalBetweenDelaysMin, long intervalBetweenDelaysMax, long delayDurationMin,
-                            long delayDurationMax) {
+    public LongGCDisruption(Random random, String disruptedNode) {
         super(random);
-        this.intervalBetweenDelaysMin = intervalBetweenDelaysMin;
-        this.intervalBetweenDelaysMax = intervalBetweenDelaysMax;
-        this.delayDurationMin = delayDurationMin;
-        this.delayDurationMax = delayDurationMax;
+        this.disruptedNode = disruptedNode;
+        this.suspendedThreadsRef = new AtomicReference<>();
     }
 
-    final static AtomicInteger thread_ids = new AtomicInteger();
-
     @Override
     public void startDisrupting() {
-        disrupting = true;
-        worker = new Thread(new BackgroundWorker(), "long_gc_simulation_" + thread_ids.incrementAndGet());
-        worker.setDaemon(true);
-        worker.start();
+        if (suspendedThreadsRef.compareAndSet(null, new HashSet<Thread>())) {
```

Comment:
can we check for null before we start and then validate it's still null on the end? an alternative is just to synchronize this block, which is simpler (but remember to do the same in stopDisrupting)